packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -669,8 +669,8 @@ export const toStyles = (
 			.has-global-padding :where(.has-global-padding) { padding-right: 0; padding-left: 0; }
 			.has-global-padding > .alignfull { margin-right: calc(var(--wp--style--root--padding-right) * -1); margin-left: calc(var(--wp--style--root--padding-left) * -1); }
 			.has-global-padding :where(.has-global-padding) > .alignfull { margin-right: 0; margin-left: 0; }
-			.has-global-padding > .alignfull:where(:not(.has-global-padding)) > :where([class*="wp-block-"]:not(.alignfull):not([class*="__"]),p,h1,h2,h3,h4,h5,h6,ul,ol) { padding-right: var(--wp--style--root--padding-right); padding-left: var(--wp--style--root--padding-left); }
-			.has-global-padding :where(.has-global-padding) > .alignfull:where(:not(.has-global-padding)) > :where([class*="wp-block-"]:not(.alignfull):not([class*="__"]),p,h1,h2,h3,h4,h5,h6,ul,ol) { padding-right: 0; padding-left: 0;`;
+			.has-global-padding > .alignfull:where(:not(.has-global-padding)) > :where(.wp-block:not(.alignfull),p,h1,h2,h3,h4,h5,h6,ul,ol) { padding-right: var(--wp--style--root--padding-right); padding-left: var(--wp--style--root--padding-left); }
+			.has-global-padding :where(.has-global-padding) > .alignfull:where(:not(.has-global-padding)) > :where(.wp-block:not(.alignfull),p,h1,h2,h3,h4,h5,h6,ul,ol) { padding-right: 0; padding-left: 0;`;
```

Comment:
Small nit: now that the PHP rules use `.wp-block` would it be worth copying the PHP rules over to the JS code here? These JS rules work perfectly fine, but I was wondering if it's worth it for the two sets of rules to be identical, so that anyone in the future looking at these rules doesn't wonder why they're different. Alternately, we could add a code comment explaining why they differ?

Reply:
I think I'd prefer to add a comment explaining the logic; it feels weird to add a useless selector just for the sake of making the rules identical. It's not impossible that they will diverge further in the future if more bugs are found 😅

Reply:
Sounds good, an explanatory comment is a reasonable way to go about it. LGTM!